### PR TITLE
Add additional variable extractors

### LINKED ISSUE #3: Increase extracting capabilities
__What should we do?__

Currently, we only extract variables from `camunda:outputParameter` properties. We should investigate whether we can increase these, e.g. by `resultVariables` from script tasks.

* [x] Create a list of (reliable) process variable sources
* [x] Update extracting mechanism by those properties
  * [x] result variables (cf. [moddle descriptors](https://github.com/camunda/camunda-bpmn-moddle/blob/master/resources/camunda.json#L422))
  * [x] form field ids (cf. [moddle descriptors](https://github.com/camunda/camunda-bpmn-moddle/blob/master/resources/camunda.json#L972))
  * [x] call activity out mappings

__Why should we do it?__

More process variables suggestions to work with.

### PR DESCRIPTION
**Pull request by pinussilvestrus**
On top of the existing `camunda:OutputParameter` extractor, this adds additional extractors for
* form fields (user tasks, start events)
* result variables (service task like elements; like business rule tasks, ....; script tasks)
* out mappings (call activities)

__Which issue does this PR address?__

Closes #3

__Acceptance Criteria__

* [ ] Corresponds to the concept 
* [ ] Corresponds to the design 

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/.github/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: src/extractors/BaseExtractor.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default function BaseExtractor() {
+
+}
+
+BaseExtractor.prototype.extractVariables = function() {
+  throw new Error('#extractVariables has to implemented.');
+};
```
Comment (barmac) [inline src/extractors/BaseExtractor.js:7]: Do we need that for anything?

File: src/extractors/CallActivityOutExtractor.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { getOutMappings } from '../util/ExtensionElementsUtil';
+
+import { createProcessVariable, addVariableToList } from '../util/ProcessVariablesUtil';
+
+
+export default function CallActivityOutExtractor() {
+  BaseExtractor.call(this);
+}
+
+CallActivityOutExtractor.extractVariables = function(options) {
```
Comment (barmac) [inline src/extractors/CallActivityOutExtractor.js:16]: Here we are using `extractVariables` as a "class method", i.e. it is not bound to instance. Anyway, do we need object here?

File: src/extractors/CallActivityOutExtractor.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      );
+
+      addVariableToList(processVariables, newVariable);
+    });
+  });
+
+  return processVariables;
+};
+
+inherits(CallActivityOutExtractor, BaseExtractor);
```
Comment (barmac) [inline src/extractors/CallActivityOutExtractor.js:50]:
```suggestion
```

We don't benefit from inheritance here, do we? :)

**Comment by barmac** [comment]
I left a few comments which apply to all of the extractors. Let's not add inheritance/objects if we don't need that. Extractors are pure functions, aren't they?

**Comment by pinussilvestrus** [comment]
Yeah, I thought at the beginning it would be a good idea to have the inheritance. But as it seems, it turns out we do not need them. I'll update it accordingly 👍

**Comment by pinussilvestrus** [comment]
I updated the PR @barmac 👍

**Comment by pinussilvestrus** [comment]
I think the little gap of test coverage is OK

**Review by barmac — APPROVED** [review]
LGTM

File: package.json [hunk 4]
```diff
@@ -36,7 +38,8 @@
     "mocha": "^8.1.1",
     "npm-run-all": "^4.1.5",
     "nyc": "^15.1.0",
-    "rollup": "^2.23.0"
+    "rollup": "^2.23.0",
+    "rollup-plugin-node-polyfills": "^0.2.1"
```
Comment (barmac) [inline package.json:42]: Why do we need this?

Comment (pinussilvestrus) [reply package.json:42]: It was somehow needed when the `inherits` dependency was included. I'll double-check whether it's needed anymore or remove it.

Comment (barmac) [reply package.json:42]: That's because `inherits` is in Node `utils` module. The only case now where we use Node module is `fs` in the tests.

**Pull request merged**